src/js/main.js

```diff
@@ -88,67 +69,217 @@ class Timelines extends React.Component {
 		}
 		else if(_.isNull(this.state.homeTimeline) && _.isNull(this.state.homeTimelineError)) {
-			homeTimelineResultOutput = " ";
-			homeTimelineResultClass = "pending";	
+			return " ";
 		}
 		else {
-			homeTimelineResultOutput = renderedTimeline(this.state.homeTimeline, true);
-			homeTimelineResultClass = "successGetTimeline";
+			return renderedTimeline(this.state.homeTimeline, true);
 		}
 	}
 
-	userTimelineResult() {
-		if(this.state.userTimelineError) {
-			userTimelineResultOutput = "There was a problem on the server side, please try again later";
-			userTimelineResultClass = "error";	
+	homeTimelineResultClass() {
+		if(this.state.homeTimelineError) {
+			return "error";	
 		}
-		else if(_.isNull(this.state.homeTimeline) && !this.state.userTimelineError) {
-			userTimelineResultOutput = " ";
-			userTimelineResultClass = "pending";	
+		else if(_.isNull(this.state.homeTimeline) && _.isNull(this.state.homeTimelineError)) {
+			return "pending";	
 		}
 		else {
-			userTimelineResultOutput = renderedTimeline(this.state.userTimeline, false);
-			userTimelineResultClass = "successGetTimeline";			
+			return "successGetTimeline";
 		}
 	}
 
 	componentDidMount() {
-		{this.handleGetUserTimeline()};
 		{this.handleGetHomeTimeline()};
-		
 	}
 
-	render(){
-		{this.homeTimelineResult()};
-		{this.userTimelineResult()};
-		return e('div', {className: 'Timelines'},[
-			e(TimelineComponent, {onKeyPressButton: this.handleKeyPress, 
+	render() {
+		{this.homeTimelineResultClass()};
+		{this.homeTimelineResultOutput()};
+		return e(TimelineComponent, {onKeyPressButton: this.handleKeyPress, 
 							 	  onChangeButton: this.handleChange, 
 								  filter: this.state.value, 
 								  key: 'homeTimelineComp', 
 							 	  timelineType: 'Home', 
 								  buttonFunc: () => {this.handleGetHomeTimeline()}, 
 								  filterButtonFunc: () => {this.handleGetFilterTimeline(this.state.value)}, 
-								  resultClass: homeTimelineResultClass, 
-								  resultOutput:homeTimelineResultOutput}, null),
-			e(TimelineComponent, {key: 'userTimelineComp', 
+								  resultClass: this.homeTimelineResultClass(), 
+								  resultOutput: this.homeTimelineResultOutput()})
+	}
+}
+
+class PostTweet extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			postTweetMessage: null,
+			postTweetError: null,
+			value: ''
+		}
+		this.handleChange = this.handleChange.bind(this);
+		this.handleKeyPress = this.handleKeyPress.bind(this);
+		this.handlePostTweet = this.handlePostTweet.bind(this);
+		this.postTweetResultClass = this.postTweetResultClass.bind(this);
+		this.postTweetResultOutput = this.postTweetResultOutput.bind(this);
+	}
+
+	handleChange(event) {
+		this.setState({
+			value: event.target.value
+		})
+	}
+	
+	handlePostTweet(message) {
+		postTweet(message).then(response => response.json())
+						  .then(data => {
+						  	this.setState ({
+						  		postTweetMessage: data.message,
+						  		postTweetError: false
```

Comment:
for boolean types, isPostTweetError would be better name
postTweetMessage being a message and postTweetError being a boolean is a bit confusion

Reply:
in general booleans using 'is<Whatever>' works well